### PR TITLE
[FLINK-33942][configuration] Fix the bug that DelegatingConfiguration misses the prefix in some get methods

### PR DESCRIPTION
**Pull request by 1996fanrui**
## What is the purpose of the change

DelegatingConfiguration misses the perfix for some methods, such as:

- DelegatingConfiguration#getInteger(org.apache.flink.configuration.ConfigOption<java.lang.Integer>, int)
- DelegatingConfiguration#getLong(org.apache.flink.configuration.ConfigOption<java.lang.Long>, long)
- DelegatingConfiguration#getBoolean(org.apache.flink.configuration.ConfigOption<java.lang.Boolean>, boolean)
- DelegatingConfiguration#getFloat(org.apache.flink.configuration.ConfigOption<java.lang.Float>, float)
- DelegatingConfiguration#getDouble(org.apache.flink.configuration.ConfigOption<java.lang.Double>, double)

## Brief change log

- [FLINK-33942][configuration][junit5-migration] Migrate DelegatingConfigurationTest to Junit5 and Assertj
- [FLINK-33942][configuration] Fix the bug that DelegatingConfiguration misses the prefix in some get methods
- [FLINK-33942][configuration][refactor] Using ConfigOption instead of string key in DelegatingConfiguration

## Verifying this change

This change added tests and can be verified as follows:

  - *DelegatingConfigurationTest#testGetWithOverrideDefault*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature?no

### CODE REVIEW

**Review by 1996fanrui — COMMENTED** [review]
Hi @RocMarshal @Sxnan , would you mind helping take a look this PR in your free time? Thanks~

File: flink-core/src/test/java/org/apache/flink/configuration/DelegatingConfigurationTest.java [hunk 1]
```diff
@@ -136,15 +136,65 @@ public void testDelegationConfigurationToMapConsistentWithAddAllToProperties() {
...
+    void testSetReturnsDelegatingConfiguration() {
         final Configuration conf = new Configuration();
         final DelegatingConfiguration delegatingConf = new DelegatingConfiguration(conf, "prefix.");
 
         Assertions.assertThat(delegatingConf.set(CoreOptions.DEFAULT_PARALLELISM, 1))
                 .isSameAs(delegatingConf);
     }
+
+    @Test
+    void testGetWithOverrideDefault() {
```
Comment (RocMarshal) [inline flink-core/src/test/java/org/apache/flink/configuration/DelegatingConfigurationTest.java:152]: I like the test case! 👍

**Review by RocMarshal — COMMENTED** [review]
Thanks @1996fanrui for catching it!
LGTM on the whole, now.
I left some minor comments.

BTW,  

if we run the followed case, will be our expected results consistent with the current fixed results?

```
    @Test
    void testUnknownCases() {
        Configuration original = new Configuration();
        final DelegatingConfiguration delegatingConf =
                new DelegatingConfiguration(original, "prefix.");

        // Test for integer
        ConfigOption<Integer> integerOption =
                ConfigOptions.key("integer.key").intType().noDefaultValue();

        // integerOption doesn't exist in delegatingConf, and it should be overrideDefault.
        original.setInteger(integerOption, 1);
        assertThat(delegatingConf.getInteger(integerOption, 2)).isEqualTo(2);

        // integerOption exists in delegatingConf, and it should be value that set before.
        delegatingConf.setInteger(integerOption, 3);
        assertThat(delegatingConf.getInteger(integerOption, 2)).isEqualTo(3);

        delegatingConf.removeConfig(integerOption);
        System.out.println(delegatingConf.get(integerOption));

    }

```

If not, it may be a bug ?

**Comment by 1996fanrui** [comment]
Thanks @RocMarshal for the quick review!
 
> if we run the followed case, will be our expected results consistent with the current fixed results?
> 
> ```
>     @Test
>     void testUnknownCases() {
>         Configuration original = new Configuration();
>         final DelegatingConfiguration delegatingConf =
>                 new DelegatingConfiguration(original, "prefix.");
> 
>         // Test for integer
>         ConfigOption<Integer> integerOption =
>                 ConfigOptions.key("integer.key").intType().noDefaultValue();
> 
>         // integerOption doesn't exist in delegatingConf, and it should be overrideDefault.
>         original.setInteger(integerOption, 1);
>         assertThat(delegatingConf.getInteger(integerOption, 2)).isEqualTo(2);
> 
>         // integerOption exists in delegatingConf, and it should be value that set before.
>         delegatingConf.setInteger(integerOption, 3);
>         assertThat(delegatingConf.getInteger(integerOption, 2)).isEqualTo(3);
> 
>         delegatingConf.removeConfig(integerOption);
>         System.out.println(delegatingConf.get(integerOption));
> 
>     }
> ```
> 
> If not, it may be a bug ?

It's indeed a bug. The `removeConfig` and `removeKey` of `DelegatingConfiguration` missed the `prefix` as well. I didn't notice them in the beginning. Would you like to fix it? If yes, feel free to take it, and I'm glad to help review.

Also, I have addressed your other comments. Please help double-check in your free time, thanks~

**Comment by RocMarshal** [comment]
> It's indeed a bug. The removeConfig and removeKey of DelegatingConfiguration missed the prefix as well. I didn't notice them in the beginning. Would you like to fix it? If yes, feel free to take it, and I'm glad to help review.

Thanks you for your confirmation . my pleasure to do it( by https://issues.apache.org/jira/browse/FLINK-33947).

**Review by RocMarshal — APPROVED** [review]
Thanks for the update.

LGTM +1 after CI successful run.

**Pull request merged**